File: livingdoc-fixture-api/src/main/java/org/livingdoc/fixture/api/converter/common/FloatConverter.java
package org.livingdoc.fixture.api.converter.common;

import java.lang.reflect.AnnotatedElement;

import org.livingdoc.fixture.api.converter.exceptions.ConversionException;
import org.livingdoc.fixture.api.converter.TypeConverter;


public class FloatConverter implements TypeConverter<Float> {

    @Override
    public Float convert(String value, AnnotatedElement element) throws ConversionException {
        try {
            return Float.valueOf(value);
        } catch (NumberFormatException e) {
            throw new ConversionException(e);
        }
    }

}

File: livingdoc-fixture-converter/src/main/java/org/livingdoc/fixture/converter/number/FloatConverter.kt
package org.livingdoc.fixture.converter.number

import java.lang.reflect.AnnotatedElement

open class FloatConverter : AbstractNumberConverter<Float>() {

    companion object {
        const val NOT_A_NUMBER = "NaN"
        const val POSITIVE_INFINITY = "Infinity"
        const val NEGATIVE_INFINITY = "-Infinity"
    }

    override fun convert(value: String, element: AnnotatedElement?): Float {
        return when (value) {
            NOT_A_NUMBER -> Float.NaN
            POSITIVE_INFINITY -> Float.POSITIVE_INFINITY
            NEGATIVE_INFINITY -> Float.NEGATIVE_INFINITY
            else -> super.convert(value, element)
        }
    }

    override fun convertToTarget(number: Number): Float = number.toFloat()

}
